manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml

```diff
@@ -15,11 +15,106 @@ metadata:
     include.release.openshift.io/ibm-cloud-managed: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
 rules:
-  - apiGroups: ["*"]
-    resources: ["*"]
-    verbs: ["*"]
   - nonResourceURLs: ["*"]
     verbs: ["*"]
+  - apiGroups:
+    - ""
+    resources:
+    - namespaces
+    - pods
+    - services
+    - pods/status
+    - configmaps
+    - secrets
+    - serviceaccounts
+    verbs:
+    - "*"
+  - apiGroups:
+    - admissionregistration.k8s.io
+    resources:
+    - mutatingwebhookconfigurations
+    - validatingwebhookconfigurations
+    verbs:
+    - "*"
+  - apiGroups:
+    - apiextensions.k8s.io
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - "*"
+  - apiGroups:
+    - apiregistration.k8s.io
+    resources:
+    - apiservices
+    verbs:
+    - "*"
+  - apiGroups:
+    - apps
+    resources:
+    - deployments
+    verbs:
+    - "*"
+  - apiGroups:
+    - batch
+    resources:
+    - jobs
+    verbs:
+    - "*"
+  - apiGroups:
+    - config.openshift.io
+    resources:
+    - clusteroperators
+    - infrastructures
+    - proxies
+    verbs:
+    - "*"
+  - apiGroups:
+    - coordination.k8s.io
+    resourceNames:
+    - packageserver-controller-lock
+    resources:
+    - leases
+    verbs:
+    - "*"
+  - apiGroups:
+    - cluster.com
+    resources:
+    - "*"
+    verbs:
+    - "*"
+  - apiGroups:
+    - operators.coreos.com
+    resources:
+    - "*"
+    verbs:
+    - "*"
+  - apiGroups:
+    - rbac.authorization.k8s.io
+    resources:
+    - clusterrolebindings
+    - clusterroles
+    - rolebindings
+    - roles
+    verbs:
+    - "*"
+  - apiGroups:
+    - example.com
+    resources:
+    - tests
+    verbs:
+    - "*"
+  - apiGroups:
+    - kiali.io
+    resources:
+    - kialis
+    verbs:
+    - "*"
+  - apiGroups:
+    - monitoring.kiali.io
+    resources:
+    - monitoringdashboards
+    verbs:
+    - "*"
```

Comment:
The RBAC proposed in this PR stemmed from a run of audit2RBAC against an e2e test's audit logs, I'm not sure how we use these RBACs.
```
- apiGroups:
    - example.com
    resources:
    - tests
    verbs:
    - "*"
  - apiGroups:
    - kiali.io
    resources:
    - kialis
    verbs:
    - "*"
  - apiGroups:
    - monitoring.kiali.io
    resources:
    - monitoringdashboards
    verbs:
    - "*"
   ```

Reply:
How confident can we be here that this audit will have captured all of the permissions we would need? I'm seeing kiali operator stuff in there and wondering if that means we would need to add every resource type from every operator that OLM manages in order to maintain functionality, which would be impractical.

Reply:
I’m not sure what other operators APIs has to do with this PR. This PR is to restrict the RBAC that is currently used by olm-operator to avoid using wildcards which means it can access a lot of other APIs that it doesn’t need. This only restricts it to a set of APIs that it needs.

Reply:
I don’t know why those kialia APIs were added in the original version of this PR but they are not needed as far as I know unless all the sudden OLM is using Kiali operator now.

Reply:
Also this PR should be on upstream as well. We can certainly refine/restrict RBAC on manifests that are used on upstream.

Reply:
Hey! So I saw a similar issue upstream and it's because of the CR/CRD validation logic that OLM has. As @dinhxuanvu pointed out, we had [an upstream PR ](https://github.com/operator-framework/operator-lifecycle-manager/pull/3076)that was a bit less strict on the RBAC we requested, I primarily was using this PR to test something on an OpenShift cluster.

The upstream PR linked earlier is being downstreamed in #600, so I'm going to close this PR.